server/storage/storage_impl.go

```diff
@@ -188,118 +306,325 @@ func (s *metaStorageImpl) PutSchemas(ctx context.Context, clusterID uint32, sche
+			}
+
+			tables = append(tables, table)
+			if table.GetId() == math.MaxUint64 {
+				log.Warn("list table table_id has reached max value", zap.Uint64("table-id", table.GetId()))
+				return tables, nil
+			}
+			nextID = table.GetId() + 1
+		}
+
+		if len(res) < rangeLimit {
+			return tables, nil
+		}
+	}
 }
 
 // TODO: operator in a batch
 func (s *metaStorageImpl) PutTables(ctx context.Context, clusterID uint32, schemaID uint32, tables []*clusterpb.Table) error {
-	for _, item := range tables {
-		key := makeTableKey(clusterID, schemaID, item.Id)
-		value, err := proto.Marshal(item)
+	for _, table := range tables {
+		key := makeTableKey(clusterID, schemaID, table.Id)
+		value, err := proto.Marshal(table)
 		if err != nil {
-			return ErrParsePutTables.WithCausef("proto parse failed, clusterID:%d, schemaID:%d, tableID:%d, err:%s", clusterID, schemaID, item.Id, err)
+			return ErrParsePutTables.WithCausef("fail to parse proto, clusterID:%d, schemaID:%d, tableID:%d, err:%v", clusterID, schemaID, table.Id, err)
 		}
 
 		if err = s.Put(ctx, key, string(value)); err != nil {
-			return errors.Wrapf(err, "meta storage put tables failed, clusterID:%d, schemaID:%d, tableID:%d, key:%s", clusterID, schemaID, item.Id, key)
+			return errors.Wrapf(err, "fail to put tables, clusterID:%d, schemaID:%d, tableID:%d, key:%s", clusterID, schemaID, table.Id, key)
 		}
 	}
 	return nil
 }
 
-// TODO: operator in a batch
-func (s *metaStorageImpl) DeleteTables(ctx context.Context, clusterID uint32, schemaID uint32, tableIDs []uint64) error {
-	for _, tableID := range tableIDs {
-		key := makeTableKey(clusterID, schemaID, tableID)
-		if err := s.Delete(ctx, key); err != nil {
-			return errors.Wrapf(err, "meta storage delete tables failed, clusterID:%d, schemaID:%d, tableID:%d, key:%s", clusterID, schemaID, tableID, key)
-		}
+func (s *metaStorageImpl) DeleteTable(ctx context.Context, clusterID uint32, schemaID uint32, tableName string) error {
+	value, err := s.Get(ctx, makeNameToIDKey(clusterID, schemaID, tableName))
+	if err != nil {
+		return errors.Wrapf(err, "fail to get table id, clusterID:%d, schemaID:%d, table name:%s", clusterID, schemaID, tableName)
+	}
+	if value == "" {
+		return ErrParseDeleteTable.WithCausef("table not found, clusterID:%d, schemaID:%d, table name:%s", clusterID, schemaID, tableName)
+	}
+	tableID, err := strconv.ParseUint(value, 10, 64)
+	if err != nil {
+		return errors.Wrapf(err, "string to int failed")
+	}
+
+	key := makeTableKey(clusterID, schemaID, tableID)
+	if err := s.Delete(ctx, key); err != nil {
+		return errors.Wrapf(err, "fail to delete table, clusterID:%d, schemaID:%d, tableName:%s, tableID:%d, key:%s", clusterID, schemaID, tableName, tableID, key)
+	}
+	if err := s.Delete(ctx, makeNameToIDKey(clusterID, schemaID, tableName)); err != nil {
+		return errors.Wrapf(err, "fail to delete table id, clusterID:%d, schemaID:%d, table name:%s", clusterID, schemaID, tableName)
```

Comment:
Maybe we should delete the two keys in a transaction to ensure them deleted at the same time.